Sources/protoc-gen-grpc-swift/Generator-Client.swift

```diff
@@ -257,3 +387,20 @@ extension MethodDescriptor {
     }
   }
 }
+
+extension Array {
+  /// Like `forEach` except that the `body` closure operates on all elements except for the last,
+  /// and the `last` closure only operates on the last element.
+  fileprivate func forEach(beforeLast body: (Element) -> (), onLast last: (Element) -> ()) {
+    var index = self.startIndex
+    while index != endIndex {
+      let element = self[index]
+      self.formIndex(after: &index)
+      if index != endIndex {
+        body(element)
+      } else {
+        last(element)
+      }
+    }
```

Comment:
Might be less difficult to write this in terms of sequence ops:

```swift
for element in self.dropLast() {
    body(element)
}
if let last = self.last {
    last(element)
}
```

Reply:
🤦‍♂️ not sure why that didn't occur to me, thanks!